src/common/io.py

```diff
@@ -1477,6 +1475,125 @@ def get_variable_data(self, name):
         else:
             return Trajectory(self._get_diagnostics_trajectory(0),factor*self._get_interpolated_trajectory(dataInd))
 
+    def get_variables_data(self,
+                           names: list[str],
+                           start_index: int = 0,
+                           stop_index: Union[int | None] = None
+    ) -> tuple[list[Trajectory], Union[int, None]]:
+        """"
+            Returns multiple trajectories, sliced to index range.
+            Note that start_index and stop_index behaves as indices for slicing, i.e. array[start_index:stop_index].
+
+            Parameters::
+
+                names --
+                    List of variables names for which to fetch trajectories.
+
+                start_index --
+                    (default: 0) Starting index for trajectory slicing.
+
+                stop_index --
+                    (default: None) Stopping index for trajectory slicing;
```

Comment:
Thinking of removing `(default: None)` since it is obvious from the type annotations now.